### DISCUSSION TITLE
`getState` is not a function in store bound via `useStore`

### DISCUSSION THREAD

**Discussion by BrooceLR** [pmndrs/zustand#1964] [Q&A]
Hello, I have created create stores in react contexts following the docs + issues, but when I try to `useMystore.getState()` I get an error `TypeError: Cannot read properties of undefined (reading 'getState')`.

Indeed, this code [from the docs](https://docs.pmnd.rs/zustand/guides/typescript#bounded-usestore-hook-for-vanilla-stores) has the same "issue" I'm having: getState does not exist on the resulting hook.

```ts
import { useStore, StoreApi } from 'zustand'
import { createStore } from 'zustand/vanilla'

interface BearState {
  bears: number
  increase: (by: number) => void
}

const bearStore = createStore<BearState>()((set) => ({
  bears: 0,
  increase: (by) => set((state) => ({ bears: state.bears + by })),
}))

const createBoundedUseStore = ((store) => (selector, equals) =>
  useStore(store, selector as never, equals)) as <S extends StoreApi<unknown>>(
  store: S
) => {
  (): ExtractState<S>
  <T>(
    selector: (state: ExtractState<S>) => T,
    equals?: (a: T, b: T) => boolean
  ): T
}

type ExtractState<S> = S extends { getState: () => infer X } ? X : never

const useBearStore = createBoundedUseStore(bearStore)
```

Attempting to access `getState`:
```ts
useBearStore.getState // TS2339: Property 'getState' does not exist on type '{ (): BearState;  (selector: (state: BearState) => T, equals?: ((a: T, b: T) => boolean) | undefined): T; }'.

```

For context, I'd like to use `getState()` in a react component that only uses the store value in a callback, so I would read it during this callback execution, rather than subscribing to it and needlessly re-render.
I've done this several times, but with global stores created via `create()`: adding react context in the mix and/or vanilla store + `useStore` seems to not allow for this.

Is this intended? Am I instead missing something?

**Comment by dai-shi** [comment]
My recommendation is to directly access `bareStore.getState`, because you already have the store.
If you would like to do `useBareStore.getState`, you could do:
```js
Object.assign(useBareStore, bareStore)
```
after of in your create function, but TS typing wouldn't be trivial.

**Comment by BrooceLR** [comment]
Thanks for taking the time, dai-shi. `Object.assign` indeed works and I had tried it already, seeing it in how `create` is [actually implemented in this repo](https://github.com/pmndrs/zustand/blob/6d9c0cff0dcb0b48027e555fd84b43d9e0cba853/src/react.ts#L88).

I think I'm trying to do something that it's just not possible: I wanted to keep the same API of a store created with `create()` for my **context-scoped** store hooks, to reduce friction for other team members.

```ts
export const useContextScopedStore = () => {
  const context = useContext(MyContext);
  if (context === undefined) {
    throw new Error("must be used within a matching context provider");
  }
  return context;
};
```

The above cannot obviously be used like `useContextScopedStore.getState()` because it *has* to be invoked to get the context 😆 

I think I'll just have two hooks, `useContextScopedStore` which returns a `StoreApi` and another one that just wraps `useStore(contextScopedStore)`

**Reply by dai-shi** [reply]
Exactly.
